### PR TITLE
chore: release 0.34.0

### PR DESCRIPTION
**Pull request by ThibaultFy**
closes FL-1246

### CODE REVIEW

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 1]
```diff
@@ -4,6 +4,93 @@
...
+Otherwise, you must update the data format. There are many ways to do this; here we give you a simple way which is acceptable if you have small amounts of data, you have access to the clusters, and downtime is acceptable. It must be repeated on each Substra node and orchestrator.
+
+#. Make sure no operations are in progress and cordon the node.
+
+#. Gather required info:
+   * database credentials (here ``$PG_USER`` and an associated password)
+   * Kubernetes namespace (here ``$NAMESPACE``)
+   * Helm release (hereafter ``$RELEASE_NAME``, can be obtained from ``helm get release -A``)
+   * hostname of the Postgres service (hereafter ``$HOST``, can be obtained from ``kubectl get svc -n $NAMESPACE``)
+   * name of the database in Postgres: by default ``orchestrator`` for the orchestrator, and ``substra`` for the backend (hereafter ``$DB``)
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:25]: Not sure that is the expected render:

image

---

```suggestion
   - database credentials (here ``$PG_USER`` and an associated password)
   - Kubernetes namespace (here ``$NAMESPACE``)
   - Helm release (hereafter ``$RELEASE_NAME``, can be obtained from ``helm get release -A``)
   - hostname of the Postgres service (hereafter ``$HOST``, can be obtained from ``kubectl get svc -n $NAMESPACE``)
   - name of the database in Postgres: by default ``orchestrator`` for the orchestrator, and ``substra`` for the backend (hereafter ``$DB``)
```

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 2]
```diff
@@ -4,6 +4,93 @@
...
+.. _ops upgrade notes 0.34:
+
+Substra 0.34.0
+--------------
+
+This version upgrades the integrated PostgreSQL databases from versions 14 to 16. This upgrade must be performed manually.
+
+If you :ref:`do not use the integrated databases <ops howto external database>`, then you have nothing to do.
+
+Otherwise, you must update the data format. There are many ways to do this; here we give you a simple way which is acceptable if you have small amounts of data, you have access to the clusters, and downtime is acceptable. It must be repeated on each Substra node and orchestrator.
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:16]: I would emphase on the requirements somehow:
```suggestion
Otherwise, you must update the data format. There are many ways to do this; here we give you a simple way which is acceptable if:
  - Have small amounts of data
  - Have access to the clusters
  - Downtime is acceptable
 It must be repeated on each Substra node and orchestrator.
```

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 3]
```diff
@@ -4,6 +4,93 @@
...
+
+#. Make sure no operations are in progress and cordon the node.
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:18]: I would emphase this sentence with a warning (not sure about the syntax tho)

```suggestion
  .. warning:: 
    Before going further, make sure no operations are in progress and cordon the node.
```

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 4]
```diff
@@ -4,6 +4,93 @@
...
+
+#. Gather required info:
+   * database credentials (here ``$PG_USER`` and an associated password)
+   * Kubernetes namespace (here ``$NAMESPACE``)
+   * Helm release (hereafter ``$RELEASE_NAME``, can be obtained from ``helm get release -A``)
+   * hostname of the Postgres service (hereafter ``$HOST``, can be obtained from ``kubectl get svc -n $NAMESPACE``)
+   * name of the database in Postgres: by default ``orchestrator`` for the orchestrator, and ``substra`` for the backend (hereafter ``$DB``)
+
+#. Deploy a psql client (we will call ``postgres-backup``) in the namespace
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:27]:
```suggestion
#. Deploy a psql client (we will call it ``postgres-backup``) in the namespace
```

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 5]
```diff
@@ -4,6 +4,93 @@
...
+      metadata:
+        name: postgres-backup
+      spec:
+        containers:
+          - name: postgres
+            image: postgres
+            command: ["sleep", "infinity"]
+      EOF
+
+#. Get a shell **onto the postgres-backup pod**, dump the DB:
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:43]: Not sure I understand the sentence as is. Here's a suggestion

```suggestion
#. Launch a shell from **within the postgres-backup pod** and dump the DB:
```

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 6]
```diff
@@ -4,6 +4,93 @@
...
+   This depends on your particular set-up but it should look like this:
+
+   .. code-block:: bash
+
+      kubectl delete -n substra statefulset $RELEASE_NAME-postgresql
+      kubectl delete -n substra pvc data-$RELEASE_NAME-postgresql-0
+
+#. Perform the database upgrade
+
+   Note versions ``8.0.0`` and ``23.0.0`` are used: they contain the database upgrade but not the app upgrade.
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:68]: I'm afraid I do not understand the sentence here.

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 7]
```diff
@@ -4,6 +4,93 @@
...
+   You can get values with ``helm get values``
+
+   orchestrator:
+
+   .. code-block:: bash
+
+      helm upgrade -n $NAMESPACE $RELEASE_NAME https://github.com/Substra/charts/raw/main/orchestrator-8.0.0.tgz --values orc-values.yaml
+
+   backend:
+   .. code-block:: bash
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:79]: Rendering issue:

image


---


```suggestion
   - Backend:
   
   .. code-block:: bash
```

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 8]
```diff
@@ -4,6 +4,93 @@
...
+      kubectl delete -n substra statefulset $RELEASE_NAME-postgresql
+      kubectl delete -n substra pvc data-$RELEASE_NAME-postgresql-0
+
+#. Perform the database upgrade
+
+   Note versions ``8.0.0`` and ``23.0.0`` are used: they contain the database upgrade but not the app upgrade.
+
+   You can get values with ``helm get values``
+
+   orchestrator:
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:72]:
```suggestion
   - Orchestrator:
```

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 9]
```diff
@@ -4,6 +4,93 @@
...
+   .. code-block:: bash
+
+      helm upgrade -n $NAMESPACE $RELEASE_NAME https://github.com/Substra/charts/raw/main/orchestrator-8.0.0.tgz --values orc-values.yaml
+
+   backend:
+   .. code-block:: bash
+
+      helm upgrade -n $NAMESPACE $RELEASE_NAME https://github.com/Substra/charts/raw/main/substra-backend-23.0.0.tgz --values backend-values.yaml
+
+   Then delete the applicative Deployments and StatefulSets to avoid them polluting the database (``orchestrator-server``, ``backend-server``, ``backend-worker``, etc)
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:83]: Feels like this could be a `#.` itself. Maybe add some commands? 

```suggestion
#.  Delete the applicative ``deployment`` and ``statefulset`` to avoid them polluting the database (``orchestrator-server``, ``backend-server``, ``backend-worker``, ...)
```

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 10]
```diff
@@ -4,6 +4,93 @@
...
+
+#. Get a shell **onto the postgres-backup pod**, load the dump:
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:85]: Same remark as before

**Review by thbcmlowk — CHANGES_REQUESTED** [review]
Requires some updates along with nitpicking remarks. Feel free to judge whether you consider them or not. Thanks for the work!

File: docs/source/how-to/deploying-substra/upgrade-notes.rst (hunk 1 above)
Comment (ThibaultFy) [reply docs/source/how-to/deploying-substra/upgrade-notes.rst:30]: Same thing in rst no ?

Comment (ThibaultFy) [reply docs/source/how-to/deploying-substra/upgrade-notes.rst:30]: good catch. I think its a blanck line missing

File: docs/source/how-to/deploying-substra/upgrade-notes.rst (hunk 6 above)
Comment (ThibaultFy) [reply docs/source/how-to/deploying-substra/upgrade-notes.rst:73]: apps upgrade are in 8.0.1 and 23.0.2

Comment (thbcmlowk) [reply docs/source/how-to/deploying-substra/upgrade-notes.rst:74]: Could we rephrase it to make it a bit clearer?

File: docs/source/how-to/deploying-substra/upgrade-notes.rst [hunk 11]
```diff
@@ -4,6 +4,102 @@
...
+#. Delete the applicative ``deployments`` and ``statefulset`` to avoid them polluting the database (``orchestrator-server``, ``backend-server``, ``backend-worker``, ...)
+
+#. Launch a shell from **within the postgres-backup pod** and load the dump:
+
+
+   .. code-block:: bash
+
+      psql --host=$HOST -U $DB_USER < /dump.sql
+
+#. Perform final upgrade as normal
```
Comment (thbcmlowk) [inline docs/source/how-to/deploying-substra/upgrade-notes.rst:101]: I feel like the [important information](https://github.com/Substra/substra-documentation/blob/chore/release-0.34.0/docs/source/how-to/deploying-substra/upgrade-notes.rst?plain=1#L22) displayed before the warning could belong here instead: 
```suggestion
#. Perform final upgrade as normal

#. Repeat on each Susbtra node, including the orchestrator (if standalone)
```

---

Else, it might need a second warning, eg something like:

```rst
  .. warning:: 
  
      It must be repeated on each Substra node and orchestrator.
```

**Review by thbcmlowk — APPROVED** [review]
Way better, thank you!
